### PR TITLE
[PTDT-1867] Add llm data gen label types

### PR DESCRIPTION
**Pull request by lgluszek**
It should be possible to run the code below to import pre-labels and ground truth

# Prompt creation

```python
import sys
import labelbox as lb
import labelbox.types as lb_types
import uuid

API_KEY=""
client = lb.Client(API_KEY, "http://localhost:8080/_gql")

prompt_annotation = lb_types.ClassificationAnnotation(
    name="Provide a prompt",
    value=lb_types.Text(answer="This is a MAL prompt 2"),
)


label = []
label.append(
  lb_types.Label(
    data=lb_types.LlmPromptCreationData(
        global_key="af3d406e-155f-451b-b828-a6f4744f2e2b"
    ),
    annotations=[
      prompt_annotation,
    ]
  )
)
  
upload_job_mal = lb.MALPredictionImport.create_from_objects(
    client = client, 
    project_id = "clptvgpf70193v2vu3xb02grt", 
    name="mal_import_job_llm_data_gen"+str(uuid.uuid4()), 
    predictions=label)
  
upload_job_mal.wait_until_done()

upload_job = lb.LabelImport.create_from_objects(
    client = client, 
    project_id = "clptvgpf70193v2vu3xb02grt", 
    name="label_import_job"+str(uuid.uuid4()),  
    labels=label)

upload_job.wait_until_done()
```

# Prompt response creation

```python
import sys
import labelbox as lb
import labelbox.types as lb_types
import uuid

API_KEY=""
client = lb.Client(API_KEY, "http://localhost:8080/_gql")


prompt_annotation = lb_types.ClassificationAnnotation(
    name="Prompt",
    value=lb_types.Text(answer="This is a MAL prompt"),
)

response_radio_annotation = lb_types.ClassificationAnnotation(
    name="Response Radio",
    value=lb_types.Radio(answer=lb_types.ClassificationAnswer(
        name="A",
        classifications=[
        lb_types.ClassificationAnnotation(
          name="Nested radio",
          value=lb_types.Radio(
            answer=lb_types.ClassificationAnswer(
              name="2"
            )
          )
        )
      ])))

response_checklist_annotation= lb_types.ClassificationAnnotation(
  name="Response Checklist", # must match your ontology feature's name
  value=lb_types.Checklist(
      answer = [
        lb_types.ClassificationAnswer(
        name="A",
        classifications=[
          lb_types.ClassificationAnnotation(
            name="Nested Checklist",
            value=lb_types.Checklist(
              answer=[
                  lb_types.ClassificationAnswer(
                    name="2"
                  )
              ]
            )
          )
        ]), 
        lb_types.ClassificationAnswer(
            name = "B"
        )
      ]
    )
 )

response_text_annotation = lb_types.ClassificationAnnotation(
    name="Response Text",
    value=lb_types.Text(answer="Look Morty, I turned myself into response text!"),
)


label = []
label.append(
  lb_types.Label(
    data=lb_types.LlmPromptResponseCreationData(
        global_key="c7a9ef3d-f417-479a-9352-629b354adbcc"
    ),
    annotations=[
        prompt_annotation,
        response_radio_annotation,
        response_checklist_annotation,
        response_text_annotation
    ]
  )
)
  
upload_job_mal = lb.MALPredictionImport.create_from_objects(
    client = client, 
    project_id = "clptu9e8c000cv2vu1nb7fii2", 
    name="mal_import_job_llm_data_gen"+str(uuid.uuid4()), 
    predictions=label)
  
upload_job_mal.wait_until_done()

upload_job = lb.LabelImport.create_from_objects(
    client = client, 
    project_id = "clptu9e8c000cv2vu1nb7fii2", 
    name="label_import_job"+str(uuid.uuid4()),  
    labels=label)

upload_job.wait_until_done()
```

# Response creation

```python
import sys
import labelbox as lb
import labelbox.types as lb_types
import uuid

API_KEY=""
client = lb.Client(API_KEY, "http://localhost:8080/_gql")

response_radio_annotation = lb_types.ClassificationAnnotation(
    name="Response Radio",
    value=lb_types.Radio(answer=lb_types.ClassificationAnswer(
        name="A",
        classifications=[
        lb_types.ClassificationAnnotation(
          name="Nested radio",
          value=lb_types.Radio(
            answer=lb_types.ClassificationAnswer(
              name="2"
            )
          )
        )
      ])))

response_checklist_annotation= lb_types.ClassificationAnnotation(
  name="Response Checklist", # must match your ontology feature's name
  value=lb_types.Checklist(
      answer = [
        lb_types.ClassificationAnswer(
        name="A",
        classifications=[
          lb_types.ClassificationAnnotation(
            name="Nested Checklist",
            value=lb_types.Checklist(
              answer=[
                  lb_types.ClassificationAnswer(
                    name="2"
                  )
              ]
            )
          )
        ]), 
        lb_types.ClassificationAnswer(
            name = "B"
        )
      ]
    )
 )

response_text_annotation = lb_types.ClassificationAnnotation(
    name="Response Text",
    value=lb_types.Text(answer="Look Morty, I turned myself into response text!"),
)


label = []
label.append(
  lb_types.Label(
    data=lb_types.LlmResponseCreationData(
        uid="cloqxnmw8007bb4vu22yf3966"
    ),
    annotations=[
        response_radio_annotation,
        response_checklist_annotation,
        response_text_annotation
    ]
  )
)
  
upload_job_mal = lb.MALPredictionImport.create_from_objects(
    client = client, 
    project_id = "clptw032p029hv2vu15su45ho", 
    name="mal_import_job_llm_data_gen"+str(uuid.uuid4()), 
    predictions=label)
  
upload_job_mal.wait_until_done()

upload_job = lb.LabelImport.create_from_objects(
    client = client, 
    project_id = "clptw032p029hv2vu15su45ho", 
    name="label_import_job"+str(uuid.uuid4()),  
    labels=label)

upload_job.wait_until_done()
```

### CODE REVIEW

File: labelbox/data/annotation_types/data/llm_prompt_creation.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from labelbox.typing_imports import Literal
+from labelbox.utils import _NoCoercionMixin
+from .base_data import BaseData
+
+
+class LlmPromptCreationData(BaseData):
```
Comment (vbrodsky) [inline labelbox/data/annotation_types/data/llm_prompt_creation.py:6]:
```suggestion
class LlmPromptCreationData(BaseData, _NoCoercionMixin):
```

File: labelbox/data/annotation_types/data/llm_prompt_response_creation.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from labelbox.typing_imports import Literal
+from labelbox.utils import _NoCoercionMixin
+from .base_data import BaseData
+
+
+class LlmPromptResponseCreationData(BaseData):
```
Comment (vbrodsky) [inline labelbox/data/annotation_types/data/llm_prompt_response_creation.py:6]: same as above

File: labelbox/data/annotation_types/data/llm_response_creation.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from labelbox.typing_imports import Literal
+from labelbox.utils import _NoCoercionMixin
+from .base_data import BaseData
+
+
+class LlmResponseCreationData(BaseData):
+    class_name: Literal["LlmResponseCreationData"] = "LlmResponseCreationData"
```
Comment (vbrodsky) [inline labelbox/data/annotation_types/data/llm_response_creation.py:7]: same as above

**Review by vbrodsky — CHANGES_REQUESTED** [review]

File: labelbox/data/annotation_types/data/llm_prompt_creation.py (hunk 1 above)
Comment (lgluszek) [reply labelbox/data/annotation_types/data/llm_prompt_creation.py]: https://github.com/Labelbox/labelbox-python/pull/1317/commits/00496afb707ab75bf813d28dace5738904f6d39a

File: labelbox/data/annotation_types/data/llm_prompt_response_creation.py (hunk 2 above)
Comment (lgluszek) [reply labelbox/data/annotation_types/data/llm_prompt_response_creation.py]: https://github.com/Labelbox/labelbox-python/pull/1317/commits/00496afb707ab75bf813d28dace5738904f6d39a

File: labelbox/data/annotation_types/data/llm_response_creation.py (hunk 3 above)
Comment (lgluszek) [reply labelbox/data/annotation_types/data/llm_response_creation.py:7]: https://github.com/Labelbox/labelbox-python/pull/1317/commits/00496afb707ab75bf813d28dace5738904f6d39a

**Review by kkim-labelbox — COMMENTED** [review]
Do we have integration tests added for these new data types?

**Comment by lgluszek** [comment]
> Do we have integration tests added for these new data types?

Added tests 6bf407381e76475720f375129748dbd7f3a8f9c2

**Pull request merged**